apps/rush-lib/src/cli/utilities/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -123,19 +123,53 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
     // version will be either:
     // A - the version (e.g. "0.0.5")
     // B - a peer dep version (e.g. "/gulp-karma/0.0.5/karma@0.13.22"
+    //                           or "/@ms/sp-client-utilities/3.1.1/foo@13.1.0"
     //                           or "/sinon-chai/2.8.0/chai@3.5.0+sinon@1.17.7")
 
     // check to see if this is the special style of specifiers
-    // e.g.:  "/gulp-karma/0.0.5/karma@0.13.22"
-    // split it by forward slashes, then grab the second group
+    // e.g.:  "/gulp-karma/0.0.5/karma@0.13.22" or
+    //     or "/@ms/sp-client-utilities/3.1.1/foo@13.1.0"
+    // split it by forward slashes, then grab the second group (or the 3rd, if the package name is scoped)
     // if the second group doesn't exist, return the version directly
     if (version) {
-      const versionParts: string[] = version.split('/');
-      if (versionParts.length !== 1 && versionParts.length !== 4) {
+      try {
+        const versionParts: string[] = version.split('/');
```

Comment:
>split [](start = 47, length = 5)

It would also be possible to use RegExps to represent these patterns:

```
^([^/]+)$
^\/[^@][^/]*\/([^/]+)
^\/\@[^/]+\/[^/]+\/([^/]+)$
```

Not sure which one is more readable